### ISSUE TITLE
Question about parallel parameter tuning best practice in PlatEMO 

### ISSUE THREAD

**Issue by gemsanyu** [BIMK/PlatEMO#74]
Hi BIMK PlatEMO team, first of all my deepest gratitude for the awesome framework.
I'd like to know if there's any example to do parameter tuning, especially in parallel just like in the GUI? Because in the manual, vectorized parameters are only for MOP not for MOEA, so parameter tuning must be done manually for each parameter setting (CMIIW?) I've tried to read the manual and the GUI modules but I still can't wrap my head around how to do it, especially I'm afraid that the GLOBAL class might be shared among parallel worker and mess up the results.

in short: i want to do a parfor loop to tune parameters and write its result to a file, is there any example on how to do it?

Thank you so much

**Comment by DestinyMy** [comment]
Hello, you can run an MOEA for multiple runs, in which different parameter values are used in different runs. For example, parameter =1 for the first ten runs, parameter=2 for the second ten runs, and so on. While in the main function of the MOEA, you can set the value of parameter as ceil(Global.run/10)

**Comment by gemsanyu** [comment]
Hi, thank you for the reply. That actually make sense but a little inconvenient to modify the algorithm as to compute the parameter values based on the run number.
I currently do it like this: 
Would you mind taking a look and tell me whether this code will do what I intend it to do and maybe tell me how to improve it?
thank you so much!

`problems = {@WFG1, @WFG2, @WFG3, @WFG4, @WFG5, @WFG6, @WFG7, @WFG8, @WFG9, ...
    @ZDT1, @ZDT2, @ZDT3, @ZDT4, @ZDT6};
probNames = {'WFG1','WFG2','WFG3','WFG4','WFG5','WFG6','WFG7','WFG8',...
    'WFG9','ZDT1','ZDT2','ZDT3','ZDT4','ZDT6'};
D = [10,10,10,10,10,10,10,10,10,30,30,30,10,30];

%ALGO params
rate = [0.01, 0.05, 0.1, 0.3, 0.5, 0.7, 1];
k = [3,4,5,7,10];
nr = [2,3,4,5,10,20,50,100];
p = 0.9;

fileID = fopen('ALGO_paramtuning.csv','w');
fprintf(fileID,'problem,m,d,p,nr,k,rate,igd,hv\n');
for pIdx = 1:14
    for rdx = 1:length(rate)
        for kdx = 1:length(k)
            for nrdx = 1:length(nr)
                igdList = zeros([1,30]);
                hvList = zeros([1,30]);
                parfor r = 1:30
                    algorithm = {@ALGO,p,nr(nrdx),rate(rdx),k(kdx)};
                    if pIdx < 10
                        setup = GLOBAL('-N',100,'-evaluation', 25000, ...
                            '-M', 2, '-D', D(pIdx), '-problem', ...
                            {problems{pIdx},8}, '-algorithm', algorithm, ...
                            '-save', 1, '-run', r);
                    else
                        setup = GLOBAL('-N',100,'-evaluation', 25000, ...
                            '-M', 2, '-D', D(pIdx), '-problem', ...
                            problems{pIdx}, '-algorithm', algorithm, ...
                            '-save', 1, '-run', r);
                    end
                    setup.Start();
                    populationObj = setup.result{2}.objs;
                    pf = setup.PF;
                    igdList(r)=IGD(populationObj, pf);
                    hvList(r)=HV(populationObj, pf);
                end
                formatf='%s,%d,%d,%8.5f,%d,%d,%8.5f,%8.5f (%8.5f), %8.5f (%8.5f)\n';
                fprintf(fileID,formatf,probNames{pIdx},2,D(pIdx),...
                    p,nr(nrdx),k(kdx),rate(rdx),mean(igdList),... 
                    std(igdList), mean(hvList), std(hvList));
            end
        end
    end
end
fclose(fileID);`
